packages/components/src/local/organisms/setting-channels/index.tsx

```diff
@@ -66,809 +33,101 @@ export const SettingChannels: React.FC<PropTypes> = ({
   channels,
   forces,
   messageTemplates,
-  selectedChannel: originalChannel
+  selectedChannel
 }) => {
-  const selectedChannelId = channels.findIndex(({ uniqid }) => uniqid === originalChannel?.uniqid)
+  const selectedChannelId = channels.findIndex(({ uniqid }) => uniqid === selectedChannel?.uniqid)
   const [selectedItem, setSelectedItem] = useState(Math.max(selectedChannelId, 0))
-  const [selectedChannelState, setSelectedChannelState] = useState<ChannelTypes | undefined>(originalChannel)
+  const [selectedChannelState, setSelectedChannelState] = useState<ChannelTypes | undefined>(selectedChannel)
   const [localChannelUpdates, setLocalChannelUpdates] = useState(channels)
   const anchorRef = useRef<HTMLDivElement>(null)
   const [open, setOpen] = useState(false)
-  const [participantKey, confirmRemoveParticipant] = useState<number>(-1)
-  const [postRemoveActionConfirmed, setPostRemoveActionConfirmed] = useState<boolean>(false)
-  const [problems, setProblems] = useState<string>('')
-  const [isChatAuthorHidden, setChatAuthorHidden] = useState<boolean>(false)
-
-  const messageTemplatesOptions: Array<Option> = messageTemplates.map(template => ({
-    name: template.title,
-    uniqid: template._id,
-    value: template
-  }))
 
   const isCollab = isCollabChannel(selectedChannelState)
-  const isChat = selectedChannelState && selectedChannelState.channelType === CHANNEL_CHAT
-  const isCustom = selectedChannelState && selectedChannelState.channelType === CHANNEL_CUSTOM
-  const isMapping = selectedChannelState && selectedChannelState.channelType === CHANNEL_MAPPING
-  const channelAsLegacy = selectedChannelState as any
-  const isLegacyCollab = channelAsLegacy && channelAsLegacy.format
-
-  // representation of channel, in more specific types
-  const mappingChannel: ChannelMapping | undefined | false = isMapping && selectedChannelState as ChannelMapping
-
-  /** init data for collab panel controls */
-  const messagesValues = getMessagesValues(isCollab, selectedChannelState)
-  const [messageLocal, setMessageLocal] = useState<MessagesValues>(messagesValues)
-
-  useEffect(() => {
-    /** on changes channel, update the message data local */
-    setMessageLocal(messagesValues)
-  }, [selectedChannelState])
+  const isChat = isChatChannel(selectedChannelState)
```

Comment:
hello @lebaphi - we have these 4 `isChat` style booleans.  But, the booleans are only used in the `renderChannelContent` method.  Instead, we could just use a `switch` inside `renderChannelContent` - and check the `channelType` value.